# PREMIERE ETAPE : OUVRIR LE FICHIER CSV

### On utilise pandas pour importer le fichier

In [42]:
import pandas as pd

### On peut aussi importer numPy

In [43]:
import numpy as np

### Import du fichier .csv

In [44]:
data = pd.read_csv("liste-des-musees-de-france-2021.csv", sep=";")

### Check du top et bottom

In [45]:
data.head(10)

osm_id                             name number  \
0  4540234591                          3 bis f    109   
1  4388745689                 A casa di Losari    NaN   
2  4426176194                 � l'homme volant     35   
3  7652598615      Abbadiale - Maison des arts    NaN   
4   121580330                           Abbaye    NaN   
5   427080450  Abbaye cistercienne de Fontenay    NaN   
6   250016053                 Abbaye d'Arthous    NaN   
7   625801255               Abbaye de Charlieu    NaN   
8   436075969   Abbaye de Clairvaux (ancienne)    NaN   
9  3024227704                  Abbaye de Cluny    NaN   

                       street  postal_code              city country  \
0  Avenue du Petit Barth�l�my      13617.0   Aix-en-Provence  France   
1                        T 30      20226.0         Belgod�re  France   
2                  Grande Rue      72300.0  Sabl�-sur-Sarthe  France   
3              Rue du Gabizos      65400.0  Arras-en-Lavedan  France   
4             Rue de l'Abbaye      39210.0    Ch�teau-Chalon  France   
5                       D 32C      21500.0          Marmagne  France   
6           Route de l'Abbaye      40300.0        Hastingues  France   
7            All�e Sunderland      42190.0          Charlieu  France   
8             Rue de l'Abbaye      10310.0         Clairvaux  France   
9       Place du 11 Ao�t 1944      71250.0             Cluny  France   

  country_code        lat       lon  \
0           fr  43.521207  5.439875   
1           fr  42.638732  9.007660   
2           fr  47.839656 -0.334043   
3           fr  42.990826 -0.128057   
4           fr  46.753866  5.624296   
5           fr  47.639979  4.390199   
6           fr  43.527247 -1.122068   
7           fr  46.157709  4.168722   
8           fr  48.146652  4.788012   
9           fr  46.434715  4.658529   

                                             website  \
0                               http://www.3bisf.com   
1                                                NaN   
2                                                NaN   
3                          https://www.abbadiale.fr/   
4                                                NaN   
5                    http://www.abbayedefontenay.com   
6                          http://arthous.landes.fr/   
7                                                NaN   
8                 https://www.abbayedeclairvaux.com/   
9  http://www.cluny.fr/71/CLUNY/TOURISME/tourisme...   

                       email         phone  fax  \
0          contact@3bisf.com  3.344216e+10  NaN   
1                        NaN           NaN  NaN   
2                        NaN  3.324395e+10  NaN   
3                        NaN  3.363317e+10  NaN   
4                        NaN           NaN  NaN   
5  info@abbayedefontenay.com  3.338092e+10  NaN   
6                        NaN  3.355873e+10  NaN   
7                        NaN           NaN  NaN   
8                        NaN           NaN  NaN   
9                        NaN  3.338559e+10  NaN   

                                      tags description        date_added  \
0          osm:museum;art:art contemporain         NaN  44116,0076388889   
1                               osm:museum         NaN  43338,9393981481   
2  osm:museum;type:centre d'interpretation         NaN  42644,4636574074   
3                               osm:museum         NaN  44120,0466782407   
4              osm:museum;art:architecture         NaN  43102,5491898148   
5              osm:museum;art:architecture         NaN  44135,4274074074   
6              osm:museum;art:architecture         NaN  44018,8086689815   
7              osm:museum;art:architecture         NaN  43523,7329976852   
8              osm:museum;art:architecture         NaN  43529,2823148148   
9              osm:museum;art:architecture         NaN   44013,637662037   

    wikidata  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN  
5    Q464918  
6   Q2820503  
7        NaN  
8  Q2

In [46]:
data.tail(10)

osm_id                                      name number  \
3774    67332571         Villa-Mus�e Jean-Honor� Fragonard    NaN   
3775  5945447856                               Villascopia     22   
3776  5396606655                              Vinopanorama    NaN   
3777    71319429                                 Visiatome    NaN   
3778  5830591853  Visite d'un r�seau de train miniature HO     17   
3779  1348524503                        Vivarium du Moulin    NaN   
3780   577287841      Voyage au Temps des impressionnistes    NaN   
3781   145062184                  Wines and Spirits Museum    NaN   
3782    82988725                                  Xanxotea    NaN   
3783   277107550          Xylotheque - Kiosque Pedagogique    NaN   

                             street  postal_code              city country  \
3774              Impasse Fragonard       6130.0            Grasse  France   
3775                Rue de Lamarque      47240.0      Castelculier  France   
3776               Route de la Cave      30420.0         Calvisson  France   
3777                          D 765      30200.0          Chusclan  France   
3778  Rue du Professeur Jean Faurel      24160.0         Excideuil  France   
3779                 Rue Principale      68610.0  Thann-Guebwiller  France   
3780                    Rue de L�ry      95430.0   Auvers-sur-Oise  France   
3781             Corniche Bonaparte      83150.0            Bandol  France   
3782                       La Place      64240.0          Isturits  France   
3783        Impasse de la Haie Vive      45590.0  Saint-Cyr-en-Val  France   

     country_code        lat       lon                            website  \
3774           fr  43.656643  6.921149    https://www.museesdegrasse.com/   
3775           fr  44.175269  0.697793                                NaN   
3776           fr  43.781922  4.196846                                NaN   
3777           fr  44.140412  4.693682                                NaN   
3778           fr  45.339207  1.044233  http://andregay1247.perso.neuf.fr   
3779           fr  47.942258  7.149613                                NaN   
3780           fr  49.072057  2.166167                                NaN   
3781           fr  43.128031  5.749484                                NaN   
3782           fr  43.364834 -1.203580           http://www.xanxotea.com/   
3783           fr  47.825888  1.967795                                NaN   

     email         phone  fax                             tags  \
3774   NaN  3.349706e+10  NaN  osm:museum;gender:musee d'homme   
3775   NaN           NaN  NaN                       osm:museum   
3776   NaN           NaN  NaN                       osm:museum   
3777   NaN           NaN  NaN                       osm:museum   
3778   NaN           NaN  NaN                       osm:museum   
3779   NaN           NaN  NaN                       osm:museum   
3780   NaN           NaN  NaN                       osm:museum   
3781   NaN           NaN  NaN                       osm:museum   
3782   NaN           NaN  NaN                       osm:museum   
3783   NaN           NaN  NaN                       osm:museum   

                                            description        date_added  \
3774                                                NaN  44176,7012615741   
3775                                                NaN  43374,9961342593   
3776                                                NaN  44166,6013773148   
3777                                                NaN  44018,3632291667   
3778                                                NaN  43325,9349189815   
3779                                                NaN  40747,8717476852   
3780                                                NaN  41141,7272222222   
3781                                                NaN  42367,7135069444   
3782  Mus�e ethnographique Xanxotea pour d�couvrir l...   44129,580775463   
3783                                          

# DEUXIEME ETAPE : QUESTIONS

## 1- Combien y-a-t-il de musées en France métropolitaine ?

### Formule .count

In [47]:
data.count()

osm_id          3784
name            3784
number           743
street          3771
postal_code     3728
city            3782
country         3784
country_code    3784
lat             3784
lon             3784
website         1636
email            301
phone           1211
fax                0
tags            3782
description      226
date_added      3775
wikidata        1147
dtype: int64

#### On peut deja voir qu'il y a 3784 donnees. Nous pouvons essayer d'etre plus precis :

### Formule len()

In [48]:
totalMuseum = len(data.index)
print("Il y a", totalMuseum , "en France metropolitaine.")

Il y a 3784 en France metropolitaine.


## 2- Dans quelle(s) ville(s) y-a-t-il de plus de musées ? 

#### Idee :  1- faire un croise dynamique qui compote le nombre de musees par ville

#### Idee 2 : prendre le max ou trier par nombre et prendre la premiere ligne

In [49]:
cityMuseum = data.pivot_table(index='city', values='name', aggfunc="count")

In [50]:
cityMuseum.head(10)

name
city                      
Abbeville                1
Abilly                   1
Ablain-Saint-Nazaire     2
Abondance                1
Agde                     2
Agen                     4
Aigremont                1
Aigues-Mortes            1
Aiguillon                1
Ainhoa                   1

#### Recherche du max (pb : afficher la ville associee)

In [51]:
nameColumn = cityMuseum["name"]
nameMax = nameColumn.max()

print("Le plus grand nombre de musees dans une ville est de",nameMax,)

Le plus grand nombre de musees dans une ville est de 175


#### Trier pour trouver le max (formule sort_values)

In [52]:
nameSorted = cityMuseum.sort_values(["name", "city"], ascending=False)

nameSorted.head(10)

name
city            
Paris        175
Lyon          33
Marseille     28
Grenoble      22
Toulouse      20
Bordeaux      20
Strasbourg    16
Nice          16
Rouen         15
Lille         15

In [53]:
cityNameMax = nameSorted.index[0]
cityMax = nameSorted.iloc[0, 0]

print(cityNameMax, cityMax)

Paris 175


In [54]:
print("La ville qui compte le plus de musees est", cityNameMax, "avec", cityMax, "unites.")

La ville qui compte le plus de musees est Paris avec 175 unites.


## 3- Quel est le nombre moyen de musées par ville ?

In [55]:
cityMean = nameSorted["name"].mean()
cityMeanRounded = np.round(cityMean, decimals = 2)

In [56]:
print("Il y a en moyenne" ,cityMeanRounded, "musee par ville.")

Il y a en moyenne 1.65 musee par ville.


## 4- Quel est le nombre médian de musées par ville ?

In [57]:
cityMedian = nameSorted["name"].median()

In [58]:
print("Le nombre median de musees par ville est de" ,cityMedian,".")

Le nombre median de musees par ville est de 1.0 .


## 5- Comment sont répartis les musées par type (en pourcentage) ?

#### Idée : colonne tags : compter les cases sans "types" ensuite garder que celles avec types, compter pour chaque type et faire pourcentage.

## 6- Combien y-a-t-il de musées dont le nom commence par "Château" ?

#### Idée : trouver la bonne formule pour trouver dans la colonne "name". Problème : comment convertir en string et comment gérer le "â" ?

Convertir en string

In [59]:
data.dtypes

osm_id            int64
name             object
number           object
street           object
postal_code     float64
city             object
country          object
country_code     object
lat             float64
lon             float64
website          object
email            object
phone           float64
fax             float64
tags             object
description      object
date_added       object
wikidata         object
dtype: object

In [60]:
data['name'].astype('str')

data.str

AttributeError: 'DataFrame' object has no attribute 'str'

In [61]:
data.str.startswith('Ch')

AttributeError: 'DataFrame' object has no attribute 'str'

## 7- Pour combien de musées dispose-t-on de l'adresse du site web ?

#### Idées : 1- pivot table et faire la somme; 2- compter les non vides dans la colonne "website"

Pivot table et somme :

In [62]:
websiteMuseum = data.pivot_table(index='name', values='website', aggfunc="count")

In [63]:
websiteMuseum.head(10)

website
name                                    
3 bis f                                1
A casa di Losari                       0
ARHome                                 1
ASM Exp�rience                         0
Abbadiale - Maison des arts            1
Abbaye                                 0
Abbaye Sainte-Odile                    0
Abbaye cistercienne de Fontenay        1
Abbaye d'Arthous                       1
Abbaye de Charlieu                     0

In [82]:
sumWebsite = websiteMuseum["website"].sum()
print("Il y a",sumWebsite, "villes dont le siteweb est renseigné.")

Il y a 1636 villes dont le siteweb est renseigné.


Compter les non vides :

## 8- Quel département français possède le plus de musées sur son territoire ?

#### Idée : même chose que par ville. Convertir d'abord dpt en string?

In [70]:
postalMuseum = data.pivot_table(index='postal_code', values='name', aggfunc="count")
postalMuseum.head(10)

name
postal_code      
1000.0          2
1100.0          2
1110.0          1
1120.0          2
1130.0          1
1150.0          1
1170.0          1
1190.0          1
1260.0          1
1340.0          1

Trier pour trouver le max

In [71]:
postalSorted = postalMuseum.sort_values(["name", "postal_code"], ascending=False)

postalSorted.head(10)

name
postal_code      
75005.0        24
38000.0        21
75007.0        16
33000.0        16
75004.0        15
76000.0        13
51100.0        13
75116.0        12
75006.0        12
75003.0        12

In [73]:
postalCodeMax = postalSorted.index[0]
PostalMax = postalSorted.iloc[0, 0]

print(postalCodeMax, PostalMax)

75005.0 24


In [75]:
print("Le département",postalCodeMax, "possède le plus grand nombre de musées avec", PostalMax, "unités.")

Le département 75005.0 possède le plus grand nombre de musées avec 24 unités.


Idée : garder seulement les deux premiers chiffres du département pour être plus précis. Enfin il faut trouver un moyen de virer le ".0"

### Remarque : les codes postaux sont à revoir. Ceux commençant par un 0 sont tronqués. Pour cette raison que cetains n'ont que 4 chiffres.

Test pour garder les deux premiers chiffres (problème : string)

In [81]:
postalSorted['dpt'] = postalSorted['postal_code'].str[:2]
postalSorted.head(10)

KeyError: 'postal_code'

## 9- Quel département français possède le moins de musées sur son territoire ?

Même chose inversé. Prob : bcp de départements auront 1 seul musées. D'où l'idée de garder seulement les deux premiers chiffres.

1- Jeter un oeil

In [78]:
postalSorted.tail(10)

name
postal_code      
1410.0          1
1400.0          1
1380.0          1
1340.0          1
1260.0          1
1190.0          1
1170.0          1
1150.0          1
1130.0          1
1110.0          1

In [79]:
postalSorted2 = postalMuseum.sort_values(["name", "postal_code"], ascending=True)

In [80]:
postalSorted2.head(10)

name
postal_code      
1110.0          1
1130.0          1
1150.0          1
1170.0          1
1190.0          1
1260.0          1
1340.0          1
1380.0          1
1400.0          1
1410.0          1

## 10- Combien de musées ont "Napoléon" dans leur nom ?

#### Idée : trouver la formule qui permet de trouver le contenu "Napoléon" dans la colonne "name". Problème : string

# 3- Proposer deux axes d'analyse supplémentaires